rootston/xdg_shell_v6.c

```diff
@@ -67,6 +67,17 @@ static void handle_request_resize(struct wl_listener *listener, void *data) {
 	view_begin_resize(input, event->cursor, view, e->edges);
 }
 
+static void handle_commit(struct wl_listener *listener, void *data) {
+	struct roots_xdg_surface_v6 *roots_xdg_surface =
+		wl_container_of(listener, roots_xdg_surface, commit);
+	struct roots_view *view = roots_xdg_surface->view;
+
+	bool centered = view_center(view);
+	if (centered) {
+		wl_list_remove(&listener->link);
+	}
```

Comment:
Why remove the link if the view is off-screen? Also, won't this center it every time the view configures itself? Like when it's resizing?

Reply:
The problem is that we get multiple `commit` events from the view before it gets a size. The goal of this is to try to center the window until it succeeds. Once the window has been centered, we remove the listener.

Reply:
I see. I think it would be better to store an "initialized" variable in the rootston struct, perhaps. We may at some point want to do more with configure events post-initialize, no?